### ISSUE TITLE
Base stats for armor missing in manifest

### ISSUE THREAD

**Issue by dad2cl3** [Bungie-net/api#238]
Looking at Devastation Complex legendary chest armor (hash = 2932121030) produces the following stats:

`    "stats": {
      "1885944937": {
        "value": 35,
        "maximum": 35,
        "minimum": 35,
        "statHash": 1885944937
      },
      "1935470627": {
        "value": 35,
        "maximum": 35,
        "minimum": 35,
        "statHash": 1935470627
      },
      "3897883278": {
        "value": 300,
        "maximum": 300,
        "minimum": 10,
        "statHash": 3897883278
      }
    }`

and investment stats:

`  "investmentStats": [
    {
      "value": 0,
      "statTypeHash": 3897883278
    },
    {
      "value": 0,
      "statTypeHash": 1935470627
    },
    {
      "value": 0,
      "statTypeHash": 1885944937
    }
  ]`

**Comment by jgayfer** [comment]
I've looked into this quite a bit. I was confused as to why most armor pieces didn't have any stats to show, when the item in game clearly has these stats. It turns out that stats can come free a few places.

1. Base stats on the armor piece
2. Stats from the item's intrinsic perk 
3. Stats from a mod/perk (choose +1 mobility or +1 recovery)

Most items in the game don't have base armor stats. Some do, definitely, but it seems as though most stats come from options 2 and 3 that I mentioned above.

I'm 90% sure this is what's happening, but I haven't looked into every item to make sure. Hope this helps.

**Comment by dad2cl3** [comment]
@jgayfer I thought all armor dropped with base stats?

**Comment by vthornheart-bng** [comment]
Yes, @jgayfer you hit the nail on the head - I'm almost certain that this is the root cause as well, I just haven't had a chance to dig into it yet.  But indeed, we had a very similar problem in D1 because armor specifically got its stat benefits from Talent Nodes and not on the base item, where weapons actually had meaningful base stats.  I'm fairly certain that your intuition on the issue is correct, and we're going to find a very similar pattern in D2 that we're not working around (yet?).

Truth be told though, there's a chance that we won't be able to give the stats if this is the case: particularly if they're coming from mods that are being assigned randomly on creation.  I'll dig in and see if I can find out.

**Comment by jgayfer** [comment]
@vthornheart-bng Yeah, sounds about right. I was pretty confused initially but after some digging I was able to account for all of my stats!

@dad2cl3 Another piece of information that might be helpful is that each class (Titan, Hunter, Warlock) has different base stats. I say this because when I was trying to map armor stats to my character stats, I was confused because my character stats were much higher than the aggregate stats on my armor.

**Comment by dad2cl3** [comment]
@jgayfer I think I might be confused. I was looking specifically at a single piece of titan chest armor. I believe(and this may be where I'm wrong) that it drops consistently with the same mobility and resilience with the possibility of a mod that might bump either mobility or resilience. As a result, I had expected that I could go into the manifest and see those base values somewhere within the definition of the armor since my recollection was that the values were the same each time the item dropped. Am I making a bad assumption?

**Comment by jgayfer** [comment]
@dad2cl3 My understanding is that item has no base stats, but gets its stats from its perks. I'm looking at the item [here](http://db.destinytracker.com/d2/en/items/2932121030-devastation-complex) by the way. It looks like `Mobile Titan Armor` gives them item some mobility, always. Then there is the option to either add some resilience or recovery via the next set of optional perks. I'm not looking at it in game though, so I can't confirm what the stats actually read as in game.

**Comment by dad2cl3** [comment]
@jgayfer thank you for the explanation!

**Comment by dad2cl3** [comment]
Alright I see the connection points now. I need to focus on the sockets and specifically the plugItemHash, when populated, for each socketTypeHash.

**Comment by vthornheart-bng** [comment]
Hmm, so I think this is actually something that would be worth bringing up for public discussion -

my gut is that, when it's defined in content that a socket will start with a specific plug, then I will go ahead and transfer any stats it provides to the item's stats in the definition.  If it's a random plug, or a plug where it is not well defined what/if a plug will be inserted, then no stats will be modified (even if some other effect that we can't predict could happen to cause a plug insertion).  This will almost certainly give us the stats that you want the vast majority (if not all) of the time, at the expense of existing users who are already examining sockets and plugs in this manner having to undo whatever they've done to work around it.

How do you all feel about that?  Does this sound good?

**Comment by jgayfer** [comment]
In the case of perks that are always active giving stats, I think it would make sense to have those show up in the item's stats in the definition.  However in the case where a perk (plug) has multiple options, I'm not sure that applying the stats provided by the default plug makes sense. Here's my thinking.

Say an item gives 1 mobility as a base stat, and 1 mobility via an intrinsic perk. In my mind, this item should have a value of 2 mobility in the definition. If we start considering the default value for optional perks, we get into a grey area. Say this same item has two perks that can be selected: the first gives 1 recovery, and the second gives 1 resilience, and the perk that gives 1 recovery is selected by default. If we use the default stats for the optional perks to set the definition, the item would have 2 mobility, and 1 recovery. However, we also see that the item has optional perks. In this scenario, it isn't clear that we are already taking into account one of the optional perks. A user could interpret it as the item having 2 mobility and 1 recovery as a base, as well as the option to add an extra resilience or recovery point via the optional perks.

That example turned out more complicated than I intended, but you get the general idea. I may have misunderstood your proposal though.

**Comment by vpzed** [comment]
Do I understand things correctly with weapons the stats in the 'stats' object of the manifest entry show the stats of the default configuration (so with the default row of perks applied and mod effect if one is applied by default with the weapon is obtained)?  Then if the player changes a perk selection, adds/changes a mod, etc. then that difference is shown in the instanced version and it could differ from the "default" stats in the manifest.

If so then I would think the armor stats would work the same way.  They reflect the stats of the default configuration of the perks (and mods if that item always comes default with it installed).

**Comment by vthornheart-bng** [comment]
Ooh, yes, this does bring up an interesting scenario... do we require app developers to know to go through each of the perks and add them up themselves, or do we require them to know the conditions under which we added it for them?

If we do what I proposed - adding the stats from sockets with default mods, or even adding the stats from sockets where only one mod is possible - then they have to know that I did that if they want to, say, show every possible combination in their UI.  They'd have to know in advance which ones we chose to pre-add, instead of being able to just iterate through all of the known mods for the socket and add it to the base.

... I think that'd be a strong reason to keep it as-is, with the stats unmodified by mods.  People would still have to know to iterate through the mods or the default mods for sockets, but there'd be slightly less to remember in the case where we leave the base item unmodified (they'd only have to remember to do that iterating, not to do that iterating but only in certain cases).

Let me know what you all think.  I'm currently leaning toward keeping it as-is given that... I'd like to err in favor of people not having to remember exceptions to the rule.

**Comment by jgayfer** [comment]
Yes, I agree with keeping it as is.

**Comment by vthornheart-bng** [comment]
Any objections or proposals from alternatives from anyone?  If not, I'll turn this into a documentation bug and add more clear documentation about what people ought to do to come up with potential/actual base stats for items under the item definition's stats property.

**Comment by dad2cl3** [comment]
No issues from me.
